src/gui/render.cpp

```diff
@@ -591,6 +591,10 @@ bool RENDER_MaybeAutoSwitchShader([[maybe_unused]] const uint16_t canvas_width,
                                   [[maybe_unused]] const bool reinit_render)
 {
 #if C_OPENGL
+	if (GFX_GetRenderingBackend() != RenderingBackend::OpenGl) {
+		return false;
+	}
+
```

Comment:
Good fix here!

Looking at the pre-processor span, I think the prior return false can be moved up here now too, and have the two texture sentinels together.

![2023-09-15_09-15](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/86c578ca-a6cd-4d9f-a4e9-2536d096d8c5)

So we can move the check for non-OpenGL usage or availability together:

```c++
#if C_OPENGL
	// We have OpenGL, but are we using it?
	if (GFX_GetRenderingBackend() != RenderingBackend::OpenGl)
# else
	// or we don't have OpenGL at all, so can't use it.
#endif
	// In both scenarios:
	{
		return false;
	}
```

There's an even cleaner alternative to the above that doesn't involve pre-processing stuff:

```c++
	if (GFX_GetRenderingBackend() == RenderingBackend::Texture) {
		return false;
	}
```

Because we only have two backends now.. but the prior approach is more robust if we add more backends.

Reply:
I agree with your above assessment and ideas @kcgen; when I originally implemented this, I was just carrying forward the ifdefs without giving them much thought, then yesterday I was in "easy in / easy out / surgical fix" mode 😄 It's great when someone else who is less close to the code makes these observations — that's where I see the value of code reviews! 🎉 

That's a good insight about removing the ifdefs from the enum itself, and with that, it turns out I can get rid of _all_ the ifdefs in `render.cpp` and convert them to runtime checks, except for a single one for conditionally compiling in the `glshader` setting. Will give it a good testing then push it as an additional commit soon.

Similar ifdef pruning could be done in `sdlmain.cpp`, but I'll leave that to @GranMinigun.

Reply:
> and with that, it turns out I can get rid of all the ifdefs in render.cpp and convert them to runtime checks

Oh wow - I didn't realize it ran that deep! Very nice taking out the werffen flammen :fire:  device to all those preprocessor branches!

Reply:
> I was just carrying forward the ifdefs without giving them much thought, then yesterday I was in "easy in / easy out / surgical fix" mode 😄 

The surgical point fixes are all good too! :man_health_worker: :scissors:... 

But I'm glad you were up for a bigger mission and took out the entire asylum .. turned into a surgical strike mission instead :laughing: